modified:   package-lock.json

slav-mirol opened the pull request:
modified:   package.json
	modified:   public/index.html
	modified:   src/App.css
	new file:   src/App.js
	deleted:    src/App.tsx
	new file:   src/components/add-movie/add-movie.css
	new file:   src/components/add-movie/add-movie.js
	new file:   src/components/header/header.css
	new file:   src/components/header/header.js
	new file:   src/components/movie-card/movie-card.css
	new file:   src/components/movie-card/movie-card.js
	new file:   src/components/movie-details/movie-details.css
	new file:   src/components/movie-details/movie-details.js
	new file:   src/components/movies-list/movie-list.css
	new file:   src/components/movies-list/movie-list.js
	modified:   src/index.css
	renamed:    src/index.tsx -> src/index.js
	deleted:    src/react-app-env.d.ts
	deleted:    src/setupTests.ts
	deleted:    tsconfig.json

slav-mirol committed:
modified:   package-lock.json
	modified:   package.json
	modified:   public/index.html
	modified:   src/App.css
	new file:   src/App.js
	deleted:    src/App.tsx
	new file:   src/components/add-movie/add-movie.css
	new file:   src/components/add-movie/add-movie.js
	new file:   src/components/header/header.css
	new file:   src/components/header/header.js
	new file:   src/components/movie-card/movie-card.css
	new file:   src/components/movie-card/movie-card.js
	new file:   src/components/movie-details/movie-details.css
	new file:   src/components/movie-details/movie-details.js
	new file:   src/components/movies-list/movie-list.css
	new file:   src/components/movies-list/movie-list.js
	modified:   src/index.css
	renamed:    src/index.tsx -> src/index.js
	deleted:    src/react-app-env.d.ts
	deleted:    src/setupTests.ts
	deleted:    tsconfig.json

slav-mirol committed:
new file:   .prettierignore
	modified:   package-lock.json
	modified:   package.json
	new file:   prettierrc.json
	modified:   src/App.css
	deleted:    src/App.js
	new file:   src/App.jsx
	modified:   src/components/add-movie/add-movie.css
	deleted:    src/components/add-movie/add-movie.js
	new file:   src/components/add-movie/add-movie.jsx
	modified:   src/components/header/header.css
	deleted:    src/components/header/header.js
	new file:   src/components/header/header.jsx
	modified:   src/components/movie-card/movie-card.css
	deleted:    src/components/movie-card/movie-card.js
	new file:   src/components/movie-card/movie-card.jsx
	modified:   src/components/movie-details/movie-details.css
	deleted:    src/components/movie-details/movie-details.js
	new file:   src/components/movie-details/movie-details.jsx
	modified:   src/components/movies-list/movie-list.css
	deleted:    src/components/movies-list/movie-list.js
	new file:   src/components/movies-list/movie-list.jsx
	modified:   src/index.css
	deleted:    src/index.js
	new file:   src/index.jsx

Kettel1 commented on src/components/add-movie/add-movie.jsx:
```diff
@@ -0,0 +1,341 @@
+import React, { useState } from "react";
+import "./add-movie.css";
+
+const AddMovie = (props) => {
```
Деструктуризируй все пропсы во всех компонентах либо про получении в параметрах, либо строкой ниже

Kettel1 commented on src/components/add-movie/add-movie.jsx:
```diff
@@ -0,0 +1,341 @@
+      <div className="form" id="add-form">
+        <p className="title">Редактирование / Создание</p>
+        <div className="params">
+          <p className="title-param">Название фильма</p>
```
У тебя повторяется структура ```p label``` вынеси это в отдельный компонент и переиспользуй это

Kettel1 commented on src/components/add-movie/add-movie.jsx:
```diff
@@ -0,0 +1,341 @@
+            placeholder="Введите название фильма"
+            value={addMovie.title}
+            onChange={(e) => {
+              setAddMovie({
```
Очень плохое решение, ты перезаписываешь весь стейт, чего делать в таком кейсе точно не стоит, а можешь сделать иначе:
```setAddMovie((prevState) => ({...prevState, posterUrl: addMovie.posterUrl}))```
ты в prevState получаешь предыдущее значаение стейта, можешь его вернуть и перезаписать нужное тебе значение

Kettel1 commented on src/components/add-movie/add-movie.jsx:
```diff
@@ -0,0 +1,341 @@
+            <p
+              className="btn-add-text"
+              onClick={() => {
+                props.addMovie(addMovie);
```
Что тут происходит?
У тебя стейт называется ```addMovie```, компонент ```AddMovie,``` и пропс ```addMovie```
Поменяй название у всего, стейт это не ```addMovie```, а ```formState```, т.к. у тебя там лежат значение формы

Kettel1 commented on src/App.jsx:
```diff
@@ -0,0 +1,123 @@
+      .then((data) => setFilms(data))
+      .catch(() => alert("Error in fetch!"));
+  }, []);
+  useEffect(() => {
```
Два одинаковых useEffect

Kettel1 commented on src/App.jsx:
```diff
@@ -0,0 +1,123 @@
+      .catch(() => alert("Error in fetch!"));
+  }
+
+  function showMovie(number) {
```
useCallback, т.к. ты эти функции передаешь в другие компоненты, а при каждом изменении стейта происходит рендер, а когда происходит рендер у тебя меняется ссылка на функцию, а когда меняется ссылка на функцию  компонент в который передана эта функция заного рендереится из-за новой ссылки

Kettel1 commented on src/App.jsx:
```diff
@@ -0,0 +1,123 @@
+    setChooseMovie(true);
+  }
+
+  function callAddMovie() {
```
useCallback

Kettel1 commented on src/App.jsx:
```diff
@@ -0,0 +1,123 @@
+    setIsAddMovie(true);
+  }
+
+  function callEditMovie() {
```
useCallback

Kettel1 commented on src/App.jsx:
```diff
@@ -0,0 +1,123 @@
+      headers: {
+        "Content-type": "application/json; charset=UTF-8",
+      },
+    })
```
Может лучше все обернуть try catch? это будет гораздо лучше чем писать столько then, + эти изменения стейта можно было положить в один then

slav-mirol committed:
modified:   server/db.json
	modified:   src/App.jsx
	modified:   src/components/add-movie/add-movie.jsx
	new file:   src/components/assets/edit-icon.jsx
	modified:   src/components/movie-card/movie-card.jsx
	modified:   src/components/movie-details/movie-details.jsx
	modified:   src/components/movies-list/movie-list.jsx

slav-mirol committed:
modified:   src/components/movies-list/movie-list.css
	modified:   src/components/movies-list/movie-list.jsx

slav-mirol committed:
modified:   src/components/movies-list/movie-list.css
	modified:   src/components/movies-list/movie-list.jsx

slav-mirol committed:
modified:   server/db.json
	modified:   src/App.jsx
	modified:   src/components/add-movie/add-movie.jsx
	modified:   src/components/movie-details/movie-details.jsx
	modified:   src/components/movies-list/movie-list.jsx

slav-mirol committed:
modified:   src/App.jsx
	modified:   src/components/add-movie/add-movie.jsx
	modified:   src/components/movie-card/movie-card.jsx
	modified:   src/components/movie-details/movie-details.jsx
	modified:   src/components/movies-list/movie-list.jsx
	new file:   src/components/movies-list/useDebounce.jsx
